src/hooks/useBalances.ts

```diff
@@ -3,22 +3,52 @@ import isEqual from 'lodash/isEqual'
 import { type SafeBalanceResponse } from '@safe-global/safe-gateway-typescript-sdk'
 import { useAppSelector } from '@/store'
 import { selectBalances } from '@/store/balancesSlice'
+import type { HiddenAssetsOnChain } from '@/store/hiddenAssetsSlice'
+import useHiddenAssets from './useHiddenAssets'
+import { safeFormatUnits, safeParseUnits } from '@/utils/formatters'
+import { BigNumber } from 'ethers'
 
-const useBalances = (): {
+const removeHiddenBalances = (
+  balances: SafeBalanceResponse,
+  hiddenAssets: HiddenAssetsOnChain,
+): SafeBalanceResponse => {
+  const fiatTotalVisible = safeFormatUnits(
```

Comment:
I wouldn't mess with the total. Hidden doesn't mean it's not there at all. I would keep the total as is sent from the backend.

Reply:
During problem shaping we made a list of all places where hiding an asset could have influence and discussed each.

For total balances we thought it would be nice as a Safe could receive scam tokens with a value denoted to them. 
Also using different token lists and manually hiding tokens would have consistent effects like this as the total balance from our tx service would also change if we filter by default list.

Reply:
When the filter is on the backend, it obviously should change the total. Tokens hidden on the client-side affecting the total makes the client-side logic too complex. We should avoid messing with the financially significant data on the client side. Besides, it creates more divergence between mobile and desktop, both technically and product-wise.
On mobile, the balance would still include the hidden tokens and it might confuse the user.